ivadomed/models.py

```diff
@@ -1430,6 +1430,212 @@ def forward(self, x):
+
+    .. seealso::
+        Alejandro Newell et al. "Stacked Hourglass Networks for Human Pose Estimation."
+        Proceedings of the European Conference on Computer Vision. 2016.
+
+    Args:
+        inplanes (int): number of input channels of the convolution kernel 
+        planes (int): number of output channels of the convolution kernel
+        stride (int): stride value in the convolution operation
+    """     
+    expansion = 2
+
+    def __init__(self, inplanes, planes, stride=1, downsample=None):
+        super(HGBottleneck, self).__init__()
+
+        self.bn1 = nn.BatchNorm2d(inplanes)
+        self.conv1 = nn.Conv2d(inplanes, planes, kernel_size=1, bias=True)
+        self.bn2 = nn.BatchNorm2d(planes)
+        self.conv2 = nn.Conv2d(planes, planes, kernel_size=3, stride=stride,
+                               padding=1, bias=True)
+        self.bn3 = nn.BatchNorm2d(planes)
+        self.conv3 = nn.Conv2d(planes, planes * 2, kernel_size=1, bias=True)
+        self.relu = nn.ReLU(inplace=True)
+        self.downsample = downsample
+        self.stride = stride
+
+    def forward(self, x):
+        residual = x
+
+        out = self.bn1(x)
+        out = self.relu(out)
+        out = self.conv1(out)
+
+        out = self.bn2(out)
+        out = self.relu(out)
+        out = self.conv2(out)
+
+        out = self.bn3(out)
+        out = self.relu(out)
+        out = self.conv3(out)
+
+        if self.downsample is not None:
+            residual = self.downsample(x)
+
+        out += residual
+
+        return out
+
+class Hourglass(nn.Module):
+    """Stacked Hourglass Networks.
+       Hourglass network is an auto-encoder decoder architecture for the task of pose estimation. In this architecture, 
+       Features are processed across all scales and consolidated to best capture the various spatial relationships associated with the pose.
+
+    .. seealso::
+        Alejandro Newell et al. "Stacked Hourglass Networks for Human Pose Estimation."
+        Proceedings of the European Conference on Computer Vision. 2016.
+
+    Args:
+        block (nn.Module): Deep bottleneck structure 
+        num_blocks (int): number of blocks in the encoder structure
+        depth (int): hourglass depth
+        name (str): model's name used for call in configuration file.
+    """    
+    def __init__(self, block, num_blocks, planes, depth):
+        super(Hourglass, self).__init__()
+        self.depth = depth
+        self.block = block
+        self.hg = self._make_hour_glass(block, num_blocks, planes, depth)
+
+    def _make_residual(self, block, num_blocks, planes):
+        layers = []
+        for i in range(0, num_blocks):
+            layers.append(block(planes*block.expansion, planes))
+        return nn.Sequential(*layers)
+
+    def _make_hour_glass(self, block, num_blocks, planes, depth):
+        hg = []
+        for i in range(depth):
+            res = []
+            for j in range(3):
```

Comment:
I would suggest storing 3 as a variable instead of passing it as a numerical value directly in the for-loop. It would make the code a bit cleaner and easier to understand/maintain

Reply:
fixed.